DOCS-8339 NetSuite Connector 10 Kim Edits (#890)
mulesoft/docs-connectors: netsuite/10.0/modules/ROOT/pages/netsuite-connector-upgrade-migrate.adoc

|9.3.0 |10.0.0
|===

== New Features and Changes in This Release

This release contains the following new features:

* These operations were added: 
** *Get Account Governance Info* 
** *Get Integration Governance Info* 

== Changed Functionality in This Release

This release contains the following changed functionality:

* The Login and SSO authentication types were removed. +
If your integration is based on these authentication types, you must migrate to a Request or Token-Based authentication type.
* Other nonfunctional changes were applied to the interface of the connector to make it more consistent with other Anypoint Connectors. +
This means that some of the operation calls in your existing application might not initially work until some minor migration tasks are performed.
* Unknown errors are now thrown as `ANY` rather than `UNKNOWN`. +

> Consequently, error codes `DISPOSE_ERROR`, `MAPPING_ERROR`, `CANT_READ_FILE`, and `TRIGGER` are now considered `ERROR_FAULT` errors instead of `UNKNOWN`.

Reviewer: I made edits to these wordings in the latest PR I edited for you...I believe it was Release notes for NetSuite Connector 10.0.0 #2474?
If you implemented those rewordings, please make these consistent with those.

Author: This actually **is** your rewording from the release notes.   So ??

Reviewer: Oh! I thought that I made changes like these:

* The Login and SSO authentication types are replaced by Request and Token-Based authentication
(or You must migrate your Login and SSO types to ....)

* Other nonfunctional changes to the connector interface might not initially work, until migration is complete.

* Unknown errors (`UNKNOWN`) are thrown as `ANY` and considered `ERROR_FAULT` errors rather than `UNKNOWN`.

I was looking for single-sentence bullets.
But what is here is fine. Whichever you prefer or think is clearest. Wasn't sure the audience needed the detail, which I expect in the doc rather than the notes.

* The *Map SSO* operation was removed.
* All exposed classes have been moved from the `org.mule.module.netsuite.extension.api` package to the `com.mulesoft.connector.netsuite.extension.api` package.

[[changed_operations]]
== Changed Operations

The behavior of some operations has changed in NetSuite Connector 10.0.

> The following operations no longer default to `payload`:

Reviewer: ```suggestion
The following operations have new default parameter values:
```

Author: They don't have new defaults. They just no longer default to payload.

Reviewer: The operations or the parameters?

Author: The parameters do not default to payload. Good catch.

[%header%autowidth.spread]
|===
|Operation | Parameter

| Change Email a| newEmail
| Change Password a| newPasword
| Get Data Center URLs a| accountId
| Get Async Result a| jobId
| Get Custom Record a| internalId 
|===

> The following operations are now *primary content* and *content* parameters:

Reviewer: ```suggestion
The following operations now take `primary content` and `content` parameters:
```
But...I see only one `content` parameter value?
Should this be worded as above, that these operations now have new default parameter values?

Author: No, they don't have new defaults

[%header%autowidth.spread]
|===
|Operation | Parameter 
| Async Search a| content 
| Async Add List a| recordAttributes 
| Async Update List a| recordAttributes 
| Async Upsert List a| recordAttributes
| Update Invitee Status List a| inviteeStatuses 
| Get Posting Transaction Summary 

> | Get Posting Transaction Summary 

Reviewer: Duplicates, without parameter values?

| Add File a| attributes 
| Add Record a| attributes  
| Get Deleted Records a| deletedDate 
| Update Record a| attributes 
| Upsert Record a| attributes 
|===

[[changed_sources]]
== Changed Sources

The three message sources were renamed:

[%header%autowidth.spread]
|===
|Source name |Old XML name | New XML name

| On New Object | `<netsuite:new-object-trigger` | `<netsuite:new-object-listener`
| On Modified Object |  `<netsuite:modified-trigger` | `<netsuite:modified-listener`
| On Deleted Object | `<netsuite:deleted-trigger` | `<netsuite:deleted-listener`
|===

== Upgrade Steps

If you are upgrading from an earlier version of the connector:

> 

Reviewer: See following suggestion.

> . A small pop-up message appears in the bottom right corner of Anypoint Studio with the `Updates Available` message.

Reviewer: This is not a step, because there is no user action in it.
Which leaves just one step...which is okay, but we can also work around it...hmmm.... Something like...:

Upgrade from an Earlier Version of the Connector

"You can now click a new `Updates Available` pop-up message in the bottom right corner of Anypoint Studio to upgrade your connector.

You can also continue to follow these steps..."

> . Click the pop-up message and follow the prompts to install the latest version.

Reviewer: See above.

You can also follow these steps to perform the upgrade:

. In Anypoint Studio, create a Mule project.
. In *Mule Palette*, click *Search in Exchange*.
. In *Add Dependencies to Project*, enter `netsuite` in the search field.
. In *Available modules*, select *NetSuite* and click *Add*. +
Studio upgrades the connector automatically.
. Click *Finish*.
. In the `pom.xml` file for the Mule project, verify that the NetSuite Connector dependency version is `10.0.0`. 

> == Additional Tasks

Reviewer: Hmm...aren't each of the following already documented above?
Did you intend a "summary" or shortcut list of actions to ugpgrading here?
Please reevaluate this text in light of that above.

Copyedits only:

> * If you were using one of the authentication schemas that were removed, you must change your configurations so that they use either *Request Based* or *Token* authentication. 

Reviewer: How do they know which were removed? Again, covered already above.
```suggestion
* Ensure that your configurations use either *Request-Based* or *Token* authentication. 
```

> * You must verify that your error handling is updated and works as expected, given the minor changes in error codes in this version of the connector. 

Reviewer: ```suggestion
* Verify that your error handling is updated to work with the minor changes in error codes in this version of the connector. 
```
Again, how would they know the minor changes? Covered earlier.

> * Refer to <<changed_operations,Changed Operations>> and <<changed_sources,Changed Sources>> for updated operations and sources. +

Reviewer: Covered earlier; these are not "additional" tasks but tasks already documented just above.

Author: Earlier they were mentioned as changes (not tasks). I changed the title of the section to "Post Upgrade Tasks," which probably makes more sense.

If your integration uses any of the affected operations and sources, make the necessary changes to comply with the new interfaces.
* All DataWeave mappings that create objects for operation inputs must be updated, since the package of the exposed classes was changed to `com.mulesoft.connector.netsuite.extension.api`. +

> You must also update mappings that transform object outputs.

Reviewer: This item is NOT covered earlier and perhaps should have its own subtopic, as do the others.

Reviewer: BREAK HERE, SECOND EDIT

Reviewer: RESUME SECOND EDIT HERE

|9.3.0 |10.0.0
|===

> == New Features and Changes in This Release

Reviewer: Do you mean to have one topic for new features, with two bullets, and then another topic about changes?
Why are both in this title if only the two list items are part of the content, and there is a following separate topic for changed functionality?

This release contains the following new features:

> * These operations were added: 

Reviewer: You want to omit this as a first-level bullet and promote the subbullets, right?
```suggestion
* *Get Account Governance Info* 
* *Get Integration Governance Info* 
```
We are calling these new features rather than new operations?

Author: Adding new operations is considered a feature. New operations = new features.

** *Get Account Governance Info* 
** *Get Integration Governance Info* 

> == Changed Functionality in This Release

Reviewer: So the previous subtitle names new features and changes (and lists two...new features? changes? both?).
This one specifies "changed functionality," which is the same or diff from "changes"?

* Other nonfunctional changes were applied to the interface of the connector to make it more consistent with other Anypoint Connectors. +
This means that some of the operation calls in your existing application might not initially work until some minor migration tasks are performed.
* Unknown errors are now thrown as `ANY` rather than `UNKNOWN`. +
Consequently, error codes `DISPOSE_ERROR`, `MAPPING_ERROR`, `CANT_READ_FILE`, and `TRIGGER` are now considered `ERROR_FAULT` errors instead of `UNKNOWN`.
* The *Map SSO* operation was removed.
* All exposed classes have been moved from the `org.mule.module.netsuite.extension.api` package to the `com.mulesoft.connector.netsuite.extension.api` package.

[[changed_operations]]
== Changed Operations

The behavior of some operations has changed in NetSuite Connector 10.0.

The following operations no longer default to `payload`:

[%header%autowidth.spread]
|===
|Operation | Parameter

| Change Email a| newEmail
| Change Password a| newPasword
| Get Data Center URLs a| accountId
| Get Async Result a| jobId
| Get Custom Record a| internalId 
|===

> The following operations are now primary content parameters:

Reviewer: The operations or the parameters?

Or "the operation parameters"?

[%header%autowidth.spread]
|===
|Operation | Parameter 
| Async Search a| criteria 
| Async Add List a| recordAttributes 
| Async Update List a| recordAttributes 
| Async Upsert List a| recordAttributes
| Update Invitee Status List a| inviteeStatuses 
| Get Posting Transaction Summary | fields 
| Get Posting Transaction Summary | filters
| Add File a| attributes 
| Add Record a| attributes  
| Get Deleted Records a| deletedDate 
| Update Record a| attributes 
| Upsert Record a| attributes 
|===

When a parameter is designated as primary content, if you do not explicitly assign a value to it, it defaults to `payload`. 

> For example, the `criteria` parameter for the Async Search operation is now designated as a primary content parameter. This means that if you do not explicitly assign a value to the `criteria` parameter, it defaults to #[payload].

Reviewer: How is `payload` different from #[payload]?
If they are both syntax, then they are both in ``.

> Additionally, since it is now a content parameter, instead of using it like this: +

Reviewer: ```suggestion
Additionally, since `criteria` is now a content parameter, you no longer use it like this: +
```

> `<netsuite:async-search config-ref="..." criteria="something">`, you now need to assign it like this: +

Reviewer: ```suggestion
`<netsuite:async-search config-ref="..." criteria="something">` +
Instead, you must assign it like this: +
```

----
<netsuite:async-search config-ref="...">
  <netsuite:criteria> something, a DW script, etc </netsuite:criteria>
</netsuite:async-search>
----

[[changed_sources]]
== Changed Sources

The three message sources were renamed:

[%header%autowidth.spread]
|===
|Source name |Old XML name | New XML name

| On New Object | `<netsuite:new-object-trigger` | `<netsuite:new-object-listener`
| On Modified Object |  `<netsuite:modified-trigger` | `<netsuite:modified-listener`
| On Deleted Object | `<netsuite:deleted-trigger` | `<netsuite:deleted-listener`
|===

== Upgrade Steps

> If you are upgrading from an earlier version of the connector, you can click the `Updates Available` pop-up message in the bottom right corner of Anypoint Studio to upgrade your connector and follow the prompts. 

Reviewer: ```suggestion
If you are upgrading from an earlier version of the connector, you can click the `Updates Available` pop-up message in the bottom right corner of Anypoint Studio and follow the prompts. 
```

You can also follow these steps to perform the upgrade:

. In Anypoint Studio, create a Mule project.
. In *Mule Palette*, click *Search in Exchange*.
. In *Add Dependencies to Project*, enter `netsuite` in the search field.
. In *Available modules*, select *NetSuite* and click *Add*. +
Studio upgrades the connector automatically.
. Click *Finish*.
. In the `pom.xml` file for the Mule project, verify that the NetSuite Connector dependency version is `10.0.0`. 

== Post Upgrade Tasks

* Ensure that your configurations use either *Request Based* or *Token* authentication. 
* Verify that your error handling is updated to work with the minor changes in error codes in this version of the connector. 
* Refer to <<changed_operations,Changed Operations>> and <<changed_sources,Changed Sources>> for updated operations and sources. +
If your integration uses any of the affected operations and sources, make the necessary changes to comply with the new interfaces.

> * All DataWeave mappings that create objects for operation inputs must be updated, since the package of the exposed classes was changed to `com.mulesoft.connector.netsuite.extension.api`. +

Reviewer: ```suggestion
* Because the package of the exposed classes was changed to `com.mulesoft.connector.netsuite.extension.api`, you must update all DataWeave mappings that create objects for operation inputs. +
```
